csrc/id_model/validation_utils.cpp

```diff
@@ -0,0 +1,138 @@
+// also considered exactly the same, IdModel maps x1 and y1, and x2
+// and y2, respectively, whereas that doesn't happen with ComputeAtMap
+//
+// Accounting for the first difference doesn't seem trivial, so when
+// swizzle is used we give up validating the exact graph. The second
+// difference is whether mappings are propagated, which can be
+// accounted for by updating the ComputeAtMap as is done in IdModel.
+void IdModelValidator::checkExactGraphEquivalence(const IdGraph& exact_graph) {
+  // Empty graph
+  if (exact_graph.disjointIdSets().disjointSets().empty()) {
+    return;
+  }
+
+  auto all_exprs = exact_graph.disjointExprSets().getAllElements();
+  if (std::find_if(all_exprs.begin(), all_exprs.end(), [](Expr* expr) {
+        return expr->isA<Swizzle2D>();
+      }) != all_exprs.end()) {
+    // Ignoring a fusion with swizzle
+    return;
+  }
+
+  Fusion* fusion = exact_graph.disjointIdSets()
+                       .disjointSets()
+                       .at(0)
+                       ->vector()
+                       .at(0)
+                       ->fusion();
+  ComputeAtMap ca_map(fusion);
+
+  DisjointSets<IterDomain*>& ca_map_exact_sets = ca_map.id_graph_.exact_nodes_;
+
+  // Propgate mappings through expressions in ComputeAtMap. Since we
+  // want to traverse and update ca_map_exact_sets, once  updated, the
+  // traversal of the ID groups cannot continue and needs to be
+  // restarted. The algorithm seems terriblly inefficient, but
+  // shuldn't matter as this is just for transitory validations
+  bool updated = true;
+  while (updated) {
+    updated = false;
+    for (const auto& set : ca_map_exact_sets.disjointSets()) {
+      auto uses = ca_map.uniqueExactUses(set->vector().front());
+      auto use_count = uses.size();
+      // Note that it should be fine to continue updating the map with
+      // the loop below as it should only modify output domain groups
+      for (size_t i = 0; i < use_count; ++i) {
+        auto use_i = uses.at(i);
+        for (size_t j = i + 1; j < use_count; ++j) {
+          auto use_j = uses.at(j);
+          if (!IterDomainGraph::exprsMap(
+                  use_i, use_j, true, ca_map_exact_sets)) {
+            continue;
+          }
+          auto num_outputs = use_i->outputs().size();
+          NVF_ERROR(use_j->outputs().size() == num_outputs);
+          for (size_t output_i = 0; output_i < num_outputs; ++output_i) {
+            auto out_i = use_i->output(output_i)->as<IterDomain>();
+            auto out_j = use_j->output(output_i)->as<IterDomain>();
+            if (!ca_map_exact_sets.strictAreMapped(out_i, out_j)) {
+              ca_map_exact_sets.mapEntries(out_i, out_j);
+              updated = true;
+            }
+          }
+        }
+      }
+      // If updated, the previous sets returned by
+      // ca_map_exact_sets.disjointSets() may contain stale sets
+      if (updated) {
+        ca_map.build(fusion);
+        break;
+      }
```

Comment:
Should we just move this code into the constructor of `ComputeAtMap`, preferably in a separate PR?

Reply:
Are you saying we should do this in `ComputeAtMap` by default?

Reply:
Yes

Reply:
That seems like a good transitory path. First we would update the ComputeAT exact map, and then the next step would replace it with the IdModel exact graph. That way, each incremental change would be minimized.

Reply:
> That seems like a good transitory path. First we would update the ComputeAT exact map, and then the next step would replace it with the IdModel exact graph. That way, each incremental change would be minimized.

And we can start this even before this PR finished reviewing. Actually, if everything goes smoothly, I would prefer to see that PR merged first, and rebase this PR on that PR.